lib/rules/dot-notation.js

```diff
@@ -59,6 +62,23 @@ module.exports = {
                             message: "[{{propertyValue}}] is better written in dot notation.",
                             data: {
                                 propertyValue: JSON.stringify(node.property.value)
+                            },
+                            fix(fixer) {
+                                const leftBracket = sourceCode.getTokenBefore(node.property);
+                                const rightBracket = sourceCode.getTokenAfter(node.property);
+                                const textBeforeProperty = sourceCode.text.slice(leftBracket.range[1], node.property.range[0]);
+                                const textAfterProperty = sourceCode.text.slice(node.property.range[1], rightBracket.range[0]);
+
+                                if (textBeforeProperty.trim() || textAfterProperty.trim()) {
+
+                                    // Don't perform any fixes if there are comments inside the brackets.
+                                    return null;
+                                }
+
+                                return fixer.replaceTextRange(
+                                    [leftBracket.range[0], rightBracket.range[1]],
+                                    `${textBeforeProperty}.${node.property.value}${textAfterProperty}`
```

Comment:
This appears to replace <code>a[&nbsp;&nbsp;'b'&nbsp;&nbsp;]</code> with <code>a&nbsp;&nbsp;.b&nbsp;&nbsp;</code>. This behaviour is not tested, and I don't think it's desirable. I would just use <code>&#x60;.${node.property.value}&#x60;</code> instead.